File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    kind = dict()
    for i in clothes:
        if kind.get(i[1]) == None:
            kind[i[1]] = 1
        else:
            kind[i[1]] += 1 
    
    answer = 1
    for i in kind.values():
        answer *= (i + 1)
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    static Map<String, Integer> map = new HashMap<>();
    public int solution(String[][] clothes) {
        int answer = 1;
        for (String[] cloth : clothes){
            if(map.get(cloth[1]) == null){
                map.put(cloth[1], 1);
            }
            else{
                map.put(cloth[1], map.get(cloth[1]) + 1);
            }
        }
        
        for (Integer num : map.values()){
            answer *= (num + 1);
        }
        
        return answer - 1;
    }
}